view/Requirements.php

```diff
@@ -1034,269 +1165,275 @@ protected function path_for_file($fileOrUrl) {
-					E_USER_WARNING);
-				return false;
-			}
-
-			// Determine if we need to build the combined include
-			if(file_exists($combinedFilePath)) {
-				// file exists, check modification date of every contained file
-				$srcLastMod = 0;
-				foreach($fileList as $file) {
-					if(file_exists($base . $file)) {
-						$srcLastMod = max(filemtime($base . $file), $srcLastMod);
+	/**
+	 * Given a set of files, combine them (as necessary) and return the url
+	 *
+	 * @param string $combinedFile Filename for this combined file
+	 * @param array $fileList List of files to combine
+	 * @return string URL to this resource
+	 */
+	protected function getCombinedFileURL($combinedFile, $fileList) {
+		// Generate path (Filename)
+		$combinedFileID = File::join_paths($this->getCombinedFilesFolder(), $combinedFile);
+
+		// Get entropy for this combined file (last modified date of most recent file)
+		$entropy = $this->getEntropyOfFiles($fileList);
+
+		// Send file combination request to the backend, with an optional callback to perform regeneration
+		$combinedURL = $this
+			->getAssetHandler()
+			->getGeneratedURL(
+				$combinedFileID,
+				$entropy,
+				function() use ($fileList) {
+					$combinedData = '';
+					$base = Director::baseFolder() . '/';
+					foreach(array_diff($fileList, $this->getBlocked()) as $file) {
+						$fileContent = file_get_contents($base . $file);
+						if ($this->writeHeaderComment) {
+							// Write a header comment for each file for easier identification and debugging.
+							$combinedData .= "/****** FILE: $file *****/\n";
+						}
+						$combinedData .= $fileContent . "\n";
 					}
+					return $combinedData;
 				}
-				$refresh = $srcLastMod > filemtime($combinedFilePath);
-			} else {
-				// File doesn't exist, or refresh was explicitly required
-				$refresh = true;
-			}
-
-			if(!$refresh) continue;
-
-			$failedToMinify = false;
-			$combinedData = "";
-			foreach(array_diff($fileList, $this->blocked) as $file) {
-				$fileContent = file_get_contents($base . $file);
+			);
 
-				try{
-					$fileContent = $this->minifyFile($file, $fileContent);
-				}catch(Exception $e){
-					$failedToMinify = true;
-				}
+		// Since url won't be automatically suffixed, add it in here
+		if($this->getSuffixRequirements()) {
+			$q = stripos($combinedURL, '?') === false ? '?' : '&';
+			$combinedURL .= "{$q}m={$entropy}";
+		}
 
-				if ($this->write_header_comment) {
-					// Write a header comment for each file for easier identification and debugging. The semicolon between each file is required for jQuery to be combined properly and protects against unterminated statements.
-					$combinedData .= "/****** FILE: $file *****/\n";
-				}
+		return $combinedURL;
+	}
 
-				$combinedData .= $fileContent . "\n";
-			}
+	/**
+	 * Check if combined files are enabled
+	 *
+	 * @return bool
+	 */
+	protected function enabledCombinedFiles() {
```

Comment:
I had two methods that did the same thing. The other was getCombinedFilesEnabled, but I'll merge them. :D